src/Math-Matrix/PMNDArrayTest.class.st

```diff
@@ -100,6 +100,53 @@ PMNDArrayTest >> testFromNestedArray [
 	
 ]
 
+{ #category : #tests }
+PMNDArrayTest >> testHadamardProduct [
+|t1 t2|
+ t1 := PMNDArray fromNestedArray:  #( 
+												#( 1 2 3 4 ) #( 5 6 7 8 ) 
+												).
+ t2 := PMNDArray fromNestedArray: #( 
+												#( 4 3 2 9 ) #( 9 7 6 5 ) 
+												).
+ self assert: (t1 hadamardProduct: t2) equals: (PMNDArray fromNestedArray:#( 
+																									#( 4 6 6 36 ) #( 45 42 42 40 ))
+																									).
+
+ t1 := PMNDArray fromNestedArray:
+				 #( 
+					#( #( 1 2 ) #( 3 4 )) 
+					#( #( 5 6 ) #( 7 8 ))
+					#( #( 9 10) #( 11 12))
+					).
+					
+ t2 := PMNDArray fromNestedArray:
+				 #( 
+					#( #( 1 3 ) #( 2 1)) 
+					#( #( 0 6 ) #( 3 8))
+					#( #( 3 5) #( 1 10))
+					).
+self assert: (t1 hadamardProduct: t2) equals: (PMNDArray fromNestedArray:
```

Comment:
Suggestion: extract the expected outcome to an intention revealing variable e.g. `expectedHadamardProduct`

Reply:
Hi @hemalvarambhia  you suggest I change `testHadamardProduct` to `expectedHadamardProduct`

Reply:
No, and I apologise for lack of clarity on my part. The `PMNDArray fromNestedArray...` part of the assertion should be extracted to a local variable called `expectedHadamardProduct`

Reply:
Thank you @hemalvarambhia but I do not really understand the interest of defining a local variable in this test

Reply:
OK. No, I understand the concerns of readability and documentation mentioned